docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,404 @@
+storage of `FieldSelector` declarations.  A `FieldSelector` supports
+a `Kind` and `APIVersion` to provide context to which object the
+selector applies against.
+
+```
+// A field selector is a field query over a set of resources. The result of matchFields and
+// matchExpressions are ANDed. An empty field selector matches all objects. A null
+// field selector matches no objects.
+type FieldSelector struct {
+  // Kind of object that scopes the selector; More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds"
+  // If omitted, the Kind is inferred from the context
+  Kind string `json:"kind,omitempty"`
+  // API version of object that scopes the selector
+  // If omitted, the API version is inferred from the context
+  APIVersion string `json:"apiVersion,omitempty"`
+  // matchFields is a map of {fieldPath,value} pairs. A single {fieldPath,value} in the matchFields
+  // map is equivalent to an element of matchExpressions, whose fieldPath field is "fieldPath", the
+  // operator is "In", and the values array contains only "value". The requirements are ANDed.
+  MatchFields map[string]string `json:"matchFields,omitempty"`
+  // matchExpressions is a list of field selector requirements. The requirements are ANDed.
+  MatchExpressions []FieldSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// A field selector requirement is a selector that contains values, a fieldPath, and an operator that
+// relates the fieldPath and values.
+type FieldSelectorRequirement struct {
+  // fieldPath is the path that the selector applies to.
+  FieldPath string `json:"fieldPath" patchStrategy:"merge" patchMergeKey:"fieldPath"`
+  // operator represents a fieldPath's relationship to a set of values.
+  // Valid operators ard In, NotIn, Exists and DoesNotExist.
+  Operator FieldSelectorOperator `json:"operator"`
+  // values is an array of string values. If the operator is In or NotIn,
+  // the values array must be non-empty. If the operator is Exists or DoesNotExist,
+  // the values array must be empty. This array is replaced during a strategic
+  // merge patch.
+  Values []string `json:"values,omitempty"`
+}
+
+// A field selector operator is the set of operators that can be used in a selector requirement.
+type FieldSelectorOperator string
+
+const (
+  FieldSelectorOpIn           FieldSelectorOperator = "In"
+  FieldSelectorOpNotIn        FieldSelectorOperator = "NotIn"
+  FieldSelectorOpExists       FieldSelectorOperator = "Exists"
+  FieldSelectorOpDoesNotExist FieldSelectorOperator = "DoesNotExist"
+)
+```
+
+### ResourceQuota
+
+The `ResourceQuotaSpec` will be modified to support the optional
+selectors to filter the `ResourceQuota` document.
+
+```
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  Hard ResourceList `json:"hard,omitempty"`
```

Comment:
IIUC ResourceList lets you specify quantities of various resources, but how do you specify a limit on the number of objects (e.g. for your "resources per namespace" and "resources per kind" use cases).